bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt

```diff
@@ -732,6 +849,47 @@ private constructor(
             this.debug = true
         }
 
+        private fun getStreamingApiUrl(instanceVersion: String?, fallbackUrl: String): String {
+            val majorVersion = instanceVersion?.substringBefore('.')
+            val version: Int = if (majorVersion == null) {
+                2
+            } else {
+                try {
+                    val majorVersionInt = majorVersion.toInt()
+                    if (majorVersionInt < 4) 1 else 2
+                } catch (e: NumberFormatException) {
+                    2
+                }
+            }
+
+            versionedInstanceRequest(version).use { response: Response ->
+                if (!response.isSuccessful) return fallbackUrl
+
+                val streamingUrl: String? = response.body?.string()?.let { responseBody: String ->
+                    val rawJsonObject = JSON_SERIALIZER
+                        .parseToJsonElement(responseBody)
+                        .jsonObject
+
+                    if (version == 2) {
+                        rawJsonObject["configuration"]
+                            ?.jsonObject
+                            ?.get("urls")
+                            ?.jsonObject
+                            ?.get("streaming") as? JsonPrimitive
+                    } else {
+                        rawJsonObject["urls"]
+                            ?.jsonObject
+                            ?.get("streaming_api") as? JsonPrimitive
+                    }?.contentOrNull
+                        // okhttp’s HttpUrl doesn't allow anything other than http(s) so we need to replace ws(s) first
+                        ?.replace("ws:", "http:")
+                        ?.replace("wss:", "https:")
```

Comment:
This might be where my confusion with code above that removes scheme prefixes is coming from. If this replacement always happens before scheme prefix removal, it should be documented like that.